pkg/recipes/configloader/secrets.go

```diff
@@ -56,32 +60,49 @@ func (e *secretsLoader) LoadSecrets(ctx context.Context, secretStoreIDResourceKe
 		}
 
 		// Populate the secretData map.
-		secretData, err = populateSecretData(secretStoreID, secretKeys, &secrets)
+		secretStoreData, err := populateSecretData(secretStoreID, secretKeysFilter, &secrets)
 		if err != nil {
 			return nil, err
 		}
+
+		// Merge the secret data for current secret store id into returned secretData map.
+		for key, value := range secretStoreData {
+			secretData[key] = value
+		}
 	}
 
 	return secretData, nil
 }
 
 // populateSecretData is a helper function to populate secret data from a secret store.
-func populateSecretData(secretStoreID string, secretKeys []string, secrets *v20231001preview.SecretStoresClientListSecretsResponse) (map[string]map[string]string, error) {
-	secretData := make(map[string]map[string]string)
-
+// It takes a secret store ID, a filter for secret keys, and a response containing the secrets.
+// If the secret keys filter is nil or empty, it retrieves data for all keys in the secret store.
+// It returns a map where the keys are the secret store IDs and the values are maps of secret keys and their corresponding values.
```

Comment:
Should start with what the function does and then give examples like you did here.